src/city/health.c

```diff
@@ -396,6 +396,12 @@ int city_health_get_global_sickness_level(void)
         }
     }
 
+    for (building *b = building_first_of_type(BUILDING_BURNING_RUIN); b; b = b->next_of_type) {
```

Comment:
You only need to check this loop if `max_sickness_level` isn't already `MAX_SICKNESS_LEVEL`.

Reply:
```
    if (max_sickness_level != MAX_SICKNESS_LEVEL) {  
        for (building *b = building_first_of_type(BUILDING_BURNING_RUIN); b; b = b->next_of_type) {
            if (b->state != BUILDING_STATE_IN_USE && b->has_plague) {
                max_sickness_level = MAX_SICKNESS_LEVEL;
                break;
            } 
        }
    }
```

Is this correct? I'm not very confident about the first `if`. :sweat_smile:

Reply:
That works, and you could use `<`instead of `!=` as well.

Welcome to programming btw!

Reply:
And... it does not work. ^^

Same behaviour as in the message above.

What I think (maybe it's stupid ):

`if (b->state != BUILDING_STATE_IN_USE && b->has_plague)` is checked only once and then it is "forgotten", because `max_sickness_level = MAX_SICKNESS_LEVEL` is only applied the moment the `if` is verified.

The first `BUILDING_BURNING_RUIN + has_plague` is found, the `if` is verified, `max_sickness_level = MAX_SICKNESS_LEVEL` is applied, and the plague message is displayed in the advisor but only at the moment the `if` is verified , then it goes back evaluating `global_sickness_level`.

But when the burning house becomes rubble, the loop is checked again, and the plague message is displayed again at the moment the `if` verified, and so on with each house becoming rubble, until the last burning house, where once again, the `max_sickness_level = MAX_SICKNESS_LEVEL` is only "triggered" the moment the `if` is verified. 

So my suggestion is to define directly `global_sickness_level` instead, when there's a burning + plagued house:

```
int global_sickness_level = SICKNESS_LEVEL_LOW;

    if (max_sickness_level < MAX_SICKNESS_LEVEL) { // no house is already plagued waiting fumigation
        for (building *b = building_first_of_type(BUILDING_BURNING_RUIN); b; b = b->next_of_type) { // check if there is a burning and plagued house
            if (b->state != BUILDING_STATE_IN_USE && b->has_plague) { 
                global_sickness_level = SICKNESS_LEVEL_PLAGUE;
                break;
            }
        }
    } else if (max_sickness_level == MAX_SICKNESS_LEVEL) { // one or many houses is plagued waiting fumigation or being fumigated
        global_sickness_level = SICKNESS_LEVEL_PLAGUE;
    } else if (max_sickness_level >= HIGH_SICKNESS_LEVEL) { // one or many houses have sickness_level >= 90
        global_sickness_level = SICKNESS_LEVEL_HIGH;
    } else if (max_sickness_level >= MEDIUM_SICKNESS_LEVEL) { // one or many houses have sickness_level >= 60
        global_sickness_level = SICKNESS_LEVEL_MEDIUM;
    }

    return global_sickness_level;
}
```
So, if there is at least one burning and plagued house, the Health advisor will display the plague message, as long the burning and plagued house exists.

This long message to finish with: "I don't know if it's possible" :sweat_smile:

Reply:
Oh god, you merged it the moment I was going to post this above!

Reply:
I thought it was working, the code looked good! Sorry there, I think you'd cash open a new PR for the same branch

Reply:
Found the issue (I think)

You have `b->state != BUILDING_STATE_IN_USE` and it should be `b->state == BUILDING_STATE_IN_USE`

Reply:
> b->state == BUILDING_STATE_IN_USE

OK, I try that first.